client/src/main/kotlin/tech/figure/classification/asset/client/client/impl/DefaultACQuerier.kt

```diff
@@ -31,6 +33,13 @@ class DefaultACQuerier(
     private val objectMapper: ObjectMapper,
     private val pbClient: PbClient,
 ) : ACQuerier {
+    private companion object {
+        // Static instance of a TypeReference for List<AssetDefinition> to prevent multiple instantiations
+        private val ASSET_DEFINITION_LIST_TYPE_REFERENCE = object : TypeReference<List<AssetDefinition>>() {}
```

Comment:
this must be the Jackson magic you referenced 😄

Reply:
Indeed lol.  You'd think it would be fine deserializing to just a list of a type from an array response, but it freaks out because it can't infer the result type due to type erasure.  Fortunately, they provide this magic TypeReference class that can maintain type information at runtime which I've always been amazed by